json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -233,6 +241,32 @@ protected function result() {
 		);
 	}
 
+	public $items = [];
+
+	public function jetpack_sync_send_data_listener() {
+		foreach ( func_get_args()[0] as $key => $item ) {
+			$items[ $key ] = $item;
+		}
+	}
+
+	public function immediate_full_sync_pull( $number_of_items ) {
+		// try to give ourselves as much time as possible.
+		set_time_limit( 0 );
+
+		remove_filter( 'jetpack_sync_send_data', array( 'Automattic\Jetpack\Sync\Actions', 'send_data' ), 10, 6 );
```

Comment:
`remove_filter` doesn't support the "number of args" param, set here to 6. You can leave that out. Also the default for the third arg is 10, so you can leave that out too.

You might also want to reduce typos and make the code more readable by making the callbacks a var, like this:

```php
$original_send_data_cb = [ 'Automattic\Jetpack\Sync\Actions', 'send_data' ];
$temp_send_data_cb = [ $this, 'jetpack_sync_send_data_listener' ];

remove_filter( 'jetpack_sync_send_data', $original_send_data_cb );
add_filter( 'jetpack_sync_send_data', $temp_send_data_cb, 10, 6 );

Sender::get_instance()->do_full_sync();

remove_filter( 'jetpack_sync_send_data', $temp_send_data_cb );
add_filter( 'jetpack_sync_send_data', $original_send_data_cb, 10, 6 );
```